targets/XPlatCppSdk/source/code/include/playfab/PlayFabAndroidHttpPlugin.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <playfab/PlayFabCallRequestContainer.h>
+#include <playfab/PlayFabPluginManager.h>
+#include <thread>
+#include <mutex>
+#include <atomic>
+
+namespace PlayFab
+{
+    /// <summary>
+    /// PlayFabAndroidHttpPlugin is an https implementation to interact with PlayFab services using WinHTTP.
+    /// </summary>
+    class PlayFabAndroidHttpPlugin : public IPlayFabHttpPlugin
+    {
+    public:
+        PlayFabAndroidHttpPlugin();
+        PlayFabAndroidHttpPlugin(const PlayFabAndroidHttpPlugin& other) = delete;
+        PlayFabAndroidHttpPlugin(PlayFabAndroidHttpPlugin&& other) = delete;
+        PlayFabAndroidHttpPlugin& operator=(PlayFabAndroidHttpPlugin&& other) = delete;
+        virtual ~PlayFabAndroidHttpPlugin();
+
+        virtual void MakePostRequest(std::unique_ptr<CallRequestContainerBase> requestContainer) override;
+        virtual size_t Update() override;
+
+    protected:
+        struct RequestTask;
+
+        void WorkerThreadEntry();
+        void WorkerThread(void* jniEnv);
+        bool ExecuteRequest(RequestTask& requestTask);
+        bool CheckResponse(RequestTask& requestTask);
+        void SetResponceAsBadRequest(RequestTask& requestTask);
+
+        virtual std::string GetUrl(RequestTask& requestTask) const;
+        virtual void SetPredefinedHeaders(RequestTask& requestTask);
+        virtual void SetHeader(RequestTask& requestTask, const char* name, const char* value);
+        virtual bool GetBinaryPayload(RequestTask& requestTask, void*& payload, size_t& payloadSize) const;
+        virtual void ProcessResponse(RequestTask& requestTask, const int httpCode);
+        virtual void HandleResults(RequestTask& requestTask);
+
+        struct RequestImpl;
+        struct RequestTask
+        {
+            RequestTask();
+            virtual ~RequestTask();
+
+            bool Initialize(std::unique_ptr<CallRequestContainerBase>& requestContainer);
+            
+            enum State:int
```

Comment:
>enum [](start = 12, length = 4)

can we use an enumclass ?

Reply:
Yes, you are right. And to use "enum class" will be better on C++, because enum class is more strictly considered type and conversion by the compiler.

---
In reply to: [273685101](https://github.com/PlayFab/SDKGenerator/pull/470#discussion_r273685101) [](ancestors = 273685101)

Reply:
to change this, it reuires many type changes. I added new backlog, and it will be better to fix once with iOS. [https://microsoft.visualstudio.com/Xbox/_workitems/edit/21252997/](https://microsoft.visualstudio.com/Xbox/_workitems/edit/21252997/)

---
In reply to: [275560116](https://github.com/PlayFab/SDKGenerator/pull/470#discussion_r275560116) [](ancestors = 275560116,273685101)